### PR TITLE
Add more about the calculation properties

### PR DESCRIPTION
**Pull request by dvirsegev**

### CODE REVIEW

File: docs/platform-overview/port-components/calculation-properties.md [hunk 1]
```diff
@@ -46,50 +54,119 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
+"calculationProperties" : {
+    "merge_config": {
+        "title": "Merge config",
+        "type": "objet",
+        "calculation": ".properties.deployed_config * .properties.service_config",
+    }
+}
+```
+
+The result will be `{cpu: 200, memory: 400}`.
```
Comment (MPTG94) [inline docs/platform-overview/port-components/calculation-properties.md:139]: Add an example of merges where the actual order has meaning (see the note block I added below), and show those two cases (similar to the example I sent you in slack earlier)

File: docs/platform-overview/port-components/calculation-properties.md [hunk 2]
```diff
@@ -46,50 +54,119 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
+
+**Notice**: This is a deep merge, The last property will **take presents** if there is any conficts of properties.
+
+:::tip
+For [Yaml](./blueprint.md#yaml) properties, the synatx of the calculation will remain the same, but the Type will be `string`
+and the format will be `yaml`.
```
Comment (MPTG94) [inline docs/platform-overview/port-components/calculation-properties.md:145]:
```suggestion
and the `format` will be `yaml`.
```

File: docs/platform-overview/port-components/calculation-properties.md [hunk 3]
```diff
@@ -46,50 +54,119 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
+    }
+}
+```
+
+The result will be `{cpu: 200, memory: 400}`.
+
+**Notice**: This is a deep merge, The last property will **take presents** if there is any conficts of properties.
+
+:::tip
+For [Yaml](./blueprint.md#yaml) properties, the synatx of the calculation will remain the same, but the Type will be `string`
```
Comment (MPTG94) [inline docs/platform-overview/port-components/calculation-properties.md]:
```suggestion
For [Yaml](./blueprint.md#yaml) properties, the synatx of the calculation will remain the same, but the `type` will be `string`
```

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please apply my suggestions, we must have an actual example of merge and override of objects, it is very hard to understand without a visual example

File: docs/platform-overview/port-components/calculation-properties.md [hunk 4]
```diff
@@ -46,50 +54,136 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
+  "type": "string",
+  "calculation": ".properties.str1 + .properties.str2"
+}
+```
+
+:::tip
+If you want to provide your own string template to concat properties with, you can do so by wrapping your template string with single quotes (`'`), for example: `'https://' + .properties.str1'`
+:::
+
+### Slice Array
```
Comment (EllaFurman) [inline docs/platform-overview/port-components/calculation-properties.md:112]:
```suggestion
### Slice array
```

File: docs/platform-overview/port-components/calculation-properties.md [hunk 5]
```diff
@@ -46,50 +54,136 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
+{
+  "title": "Slice array example",
+  "type": "string",
+  "calculation": ".properties.array1[1:4]"
+}
+```
+
+### Merge Objects
+
+Assume you have two `object` properties: one called `deployed_config` with the value `{cpu: 200}`, the other called `service_config` with the value `{memory: 400}`,and You can merge these two object properties and get a unified config by using the following calculation:
```
Comment (EllaFurman) [inline docs/platform-overview/port-components/calculation-properties.md:126]:
```suggestion
### Merge objects

Assume you have two `object` properties: one called `deployed_config` with the value `{cpu: 200}`, the other called `service_config` with the value `{memory: 400}`. You can merge these two object properties and receive a unified config by using the following calculation:
```

File: docs/platform-overview/port-components/calculation-properties.md [hunk 6]
```diff
@@ -46,50 +54,136 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
 
-Then the `merge_config` Calculation Property value will be:
+The result will be `{cpu: 200, memory: 400}`.
+
+:::info Object merging
+
+- Object merging performs a deep merge, nested keys from the original objects will also appear in the resulting merged object.
+- If the same `key` appears in one or more of the merged properties, the last property that appears will have its `keys` take precedence over the `keys` of properties that appeared earlier in the calculation.
+
+For example, Let's assume we have 2 properties with type `object`, and we want to perform a deep merge between them:
```
Comment (EllaFurman) [inline docs/platform-overview/port-components/calculation-properties.md:145]:
```suggestion
:::Info object merging

- Object merging performs a deep merge, resulting in nested keys from the original objects appearing in the resulting merged object.
- If the same `key` appears in one or more of the merged properties, the last property that appears will have its `keys` take precedence over the `keys` of properties that appeared earlier in the calculation.

For example, Let's assume we have 2 properties with type `object`, and we want to perform a deep merge between them:
```

File: docs/platform-overview/port-components/calculation-properties.md [hunk 7]
```diff
@@ -46,50 +54,136 @@ Inside the Calculation Property object you can specify the `title` to grant the
...
+
+If the calculation is `".properties.obj2 * .properties.obj1"` , the result will be `{cpu: 200}`,
+
+For merging YAML properties, the merging behavior will be the same, but if you specify `type: "string` and `format: "yaml"`, the result will be a YAML object.
+:::
+
+## Calculation property Edge cases
+
+In some occasions, if the key contains special characters or starts with a digit, you need to surround it with double quotes like this: .`"foo$"`
+For example, if you want to use your `on-call` property in a Calculation property (note the single quotes (`'`) around `on-call`):
```
Comment (EllaFurman) [inline docs/platform-overview/port-components/calculation-properties.md:168]:
```suggestion
## Calculation Property edge cases

Sometimes, if the key contains special characters or starts with a digit, you need to surround it with double quotes like this: `"foo$"`.
For example, if you want to use your `on-call` property in a Calculation Property (note the single quotes (`'`) around `on-call`):
```

**Review by EllaFurman — CHANGES_REQUESTED** [review]

File: docs/platform-overview/port-components/calculation-properties.md [hunk 8]
```diff
@@ -7,7 +7,14 @@ sidebar_position: 7
...
 Calculation Properties make it easier to define properties that are based on values from other properties, with the added ability to transform the data.
-Calculation Properties can be used for merging, slicing, selecting values, etc.
+
+The Calculation Properties support the types `string`, `number`, `object`, `array`, `boolean`, `yaml` .
+
+With Calculation Properties you can:
+
+- Filter/Select/Slice/Concat from an existing property.
+- Create math equations or modification. For example - calculate required disk storage by specifying page size, and number of pages needed.
+- Merge complex properties, including deep-merge and overriding.
```
[the remark this answers is not in the document]
Comment (dvirsegev) [reply docs/platform-overview/port-components/calculation-properties.md:17]: On purpose,
(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat)

**Review by MPTG94 — APPROVED** [review]
LGTM

**Pull request merged**